### PR TITLE
Template Refactor

### LINKED ISSUE #8: Update packaging of template
Rename the "root" source folder from "otio_plugin_template" -> "src" as suggested by @JeanChristopheMorinPerso for more robust testing and avoid faulty import of uninstalled code.

### PR DESCRIPTION
**Pull request by apetrynet**
The goal of this PR is to create a cleaner template with an updated packaging strategy. 
Closes #8  

Highlights:
* pyproject.toml replaces setup.py/cfg
* Updated folder structure
* More consistent naming of project and folder names

### CODE REVIEW

**Comment by apetrynet** [comment]
I need to update the wheel build

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
```
Comment (JeanChristopheMorinPerso) [inline pyproject.toml:3]: :+1:

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Looks great! I love that it now uses hatchling instead of setuptools!

We might want to specify that the project can be built locally by using https://pypa-build.readthedocs.io/en/stable/.

File: pyproject.toml (hunk 1 above)
Comment (apetrynet) [reply pyproject.toml:3]: I just followed your lead 🙂

Comment (JeanChristopheMorinPerso) [reply pyproject.toml:3]: So you used https://github.com/JeanChristopheMorinPerso/otio-fcpxxml-adapter/blob/initial_structure/pyproject.toml as a reference?

Comment (apetrynet) [reply pyproject.toml:3]: Yes.

Comment (JeanChristopheMorinPerso) [reply pyproject.toml:3]: Great, I'm happy that it was useful!

Comment (apetrynet) [reply pyproject.toml:3]: Added a segment on building packages.

File: README.md [hunk 2]
```diff
@@ -144,6 +147,18 @@ otioconvert -i some_timeline.otio -o some_timeline.ext
...
+### Build your package locally
+You might want to build your package locally to check that everything is behaving
+the way you intended.  
+To do so, simply run `python -m build` in the root of your repo.  
+This will by default produce wheel and source packages in a "dist" folder.
+>**NOTE!** You might need to add the "build" package to your virtualenv (`pip install build`).
+
+For more info on building packages, please refer to the python's 
+[packaging](https://packaging.python.org/en/latest/overview/) guide 
+and/or the [build](https://pypa-build.readthedocs.io/en/stable/) documentation
```
Comment (JeanChristopheMorinPerso) [inline README.md:159]:
```suggestion
and/or the [build](https://pypa-build.readthedocs.io/en/stable/) documentation.
```

**Pull request merged**